### PR TITLE
Add `--max-processes` option

### PR DESCRIPTION
**Pull request by HypeMC**
This PR adds a `--max-processes` option that, when used with `--processes=auto`, limits the number of processes so it does not exceed a specified maximum. This is useful when the number of CPU cores on CI servers varies, but you don't want to use all of them in some cases.

### CODE REVIEW

File: test/Unit/OptionsTest.php [hunk 1]
```diff
@@ -49,6 +50,13 @@ public function testAutoProcessesMode(): void
         self::assertEquals(Options::getNumberOfCPUCores(), $options->processes);
     }
 
+    public function testAutoProcessesModeWithMaxProcesses(): void
+    {
+        $options = $this->createOptionsFromArgv(['--processes' => 'auto', '--max-processes' => '3']);
+
+        self::assertEquals(min(Options::getNumberOfCPUCores(), 3), $options->processes);
```
Comment (Slamdunk) [inline test/Unit/OptionsTest.php]:
```suggestion
        $options = $this->createOptionsFromArgv(['--processes' => 'auto', '--max-processes' => '0']);

        self::assertSame(0, $options->processes);
```

Copy-pasting part of the `src/` code defeats the purpose of a test. We can have a more effective test by pushing the boundaries on non-so meaningfull cases

**Review by Slamdunk — CHANGES_REQUESTED** [review]

File: test/Unit/OptionsTest.php (hunk 1 above)
Comment (HypeMC) [reply test/Unit/OptionsTest.php]: Done

**Review by Slamdunk — APPROVED** [review]
